src/components/Single-Article.jsx

```diff
@@ -48,6 +61,30 @@ const SingleArticle = (props) => {
     });
   }
 
+  function handleDeletion(e){
+
+const text = e.target.innerText
+
+let comment_id = Number(text.slice(-3))
+
+
+    removeComment(comment_id).then((result) =>{ if (result) { 
+      alert("Succesfully deleted comment :)");
+    } 
+  }).catch((err) => { 
+    alert("Something went wrong ):, please try again.") 
+  }); 
+
+
+    const filteredComments =  comments.filter((commentToDelete) => { 
+      return commentToDelete.comment_id !== comment_id
+    }); 
+
+    setComments([...filteredComments])
+
```

Comment:
set comments here will render the comments optimistically. The deleted comment will disappear straight away even if there's an error when the api call is made.  We probably want to remove the comment after the deleted request is successfully made. Moving line79 to 83 into the .then() of removeComment will fix the issue